test/functional/test_framework/util.py

```diff
@@ -47,20 +47,44 @@ def assert_fee_amount(fee, tx_size, feerate_BTC_kvB):
         raise AssertionError("Fee of %s BTC too high! (Should be %s BTC)" % (str(fee), str(target_fee)))
 
 
-def assert_equal(thing1, thing2, *args):
+def assert_equal(thing1, thing2, *args, err_msg=None):
     if thing1 != thing2 or any(thing1 != arg for arg in args):
-        raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
+        msg = "not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args)
+        if err_msg is not None:
+            msg = err_msg
+        raise AssertionError(msg)
```

Comment:
IMHO the only reason your solution is better than mine is that you avoid the copy of the string, otherwise is only a different way to a boolean condition :)

Reply:
Oh yeah I didn't mean to make it sound like a big thing, just that in this order it's a bit less code that executes a bit faster. Thanks for updating!

Reply:
Looks like you only updated it for `assert_true` but it holds for all the other functions to which you added `err_msg` too.

Reply:
ops :) sorry too lazy! I will update the the others in a few hours :) Thanks for the catch